### ISSUE TITLE
Color Decision List (CDL) support

### ISSUE THREAD

**Issue by jminor** [PixarAnimationStudios/OpenTimelineIO#142]
Some people would like OTIO to handle Color Decision List metadata in a structured way.

For example, an EDL can contain ASC_SOP and ASC_SAT like this:
*ASC_SOP (1.0000 1.0000 1.0000)(0.0000 0.0122 0.0305)(1.0000 1.0000 1.0000)
*ASC_SAT 1.0000

For reference:
https://en.wikipedia.org/wiki/ASC_CDL

**Comment by jminor** [comment]
@adaro can you take a look at this to see if it parses your CDL data correctly?

**Comment by jminor** [comment]
CDL metadata is now supported in the EDL adapter (both read and write).

CDL metadata on a Clip in OTIO looks like this:
```
                        "metadata": {
                            "cdl": {
                                "asc_sat": 0.9, 
                                "asc_sop": {
                                    "offset": [
                                        1.0, 
                                        0.0122, 
                                        0.0305
                                    ], 
                                    "power": [
                                        1.0, 
                                        0.0, 
                                        1.0
                                    ], 
                                    "slope": [
                                        0.1, 
                                        0.2, 
                                        0.3
                                    ]
                                }
                            }
                         }
```

We still need to add support for CDL to the FCP XML adapter.

**Comment by jminor** [comment]
Here is a reference for how CDL data is stored in AAF (as a Video Effect).
https://www.amwa.tv/downloads/specifications/AS-05_AAF_Effects_protocol_v1.pdf

**Comment by jminor** [comment]
Here is a reference for how CDL data is stored in FCP X XML (as an XML comment inside a color correction video filter).
https://developer.apple.com/library/content/documentation/FinalCutProX/Reference/FinalCutProXXMLFormat/Adjustments/Adjustments.html#//apple_ref/doc/uid/TP40011227-CH14-SW8

**Comment by jminor** [comment]
I think CDL support in OTIO would be a great candidate for expanding our support for effects. It really is a color correction effect with well defined parameters, so we can model it as a schema.Effect object on a Clip. Each adapter can then transcribe that to/from the various other formats as appropriate.

**Comment by ssteinbach** [comment]
Addressed by #143
